File: Src/Icm.Core/Tuples/ModifiableTuplesFactory.vb
Namespace Icm
    Public Module ModifiableTuplesFactory

        Public Function NewPair(Of T1, T2)(ByVal o1 As T1, ByVal o2 As T2) As Pair(Of T1, T2)
            Return New Pair(Of T1, T2)(o1, o2)
        End Function

        Public Function NewTuple3(Of T1, T2, T3)(ByVal o1 As T1, ByVal o2 As T2, ByVal o3 As T3) As Tuple3(Of T1, T2, T3)
            Return New Tuple3(Of T1, T2, T3)(o1, o2, o3)
        End Function

        Public Function NewTuple4(Of T1, T2, T3, T4)(ByVal o1 As T1, ByVal o2 As T2, ByVal o3 As T3, ByVal o4 As T4) As Tuple4(Of T1, T2, T3, T4)
            Return New Tuple4(Of T1, T2, T3, T4)(o1, o2, o3, o4)
        End Function


    End Module

End Namespace

File: Src/Icm.Core/Tuples/ModifiableTuplesFactory.cs
namespace Icm
{
	public static class ModifiableTuplesFactory
	{

		public static Pair<T1, T2> NewPair<T1, T2>(T1 o1, T2 o2)
		{
			return new Pair<T1, T2>(o1, o2);
		}

		public static Tuple3<T1, T2, T3> NewTuple3<T1, T2, T3>(T1 o1, T2 o2, T3 o3)
		{
			return new Tuple3<T1, T2, T3>(o1, o2, o3);
		}

		public static Tuple4<T1, T2, T3, T4> NewTuple4<T1, T2, T3, T4>(T1 o1, T2 o2, T3 o3, T4 o4)
		{
			return new Tuple4<T1, T2, T3, T4>(o1, o2, o3, o4);
		}


	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
